frontend/src/App.test.js

```diff
@@ -25,12 +26,24 @@ beforeAll(() => {
   global.Date.now = mockedDate.now;
 });
 
+jest.mock('axios');
+
 describe('App', () => {
   it('should be defined', () => {
     expect(App).toBeDefined();
   });
+
   it('should render correctly', () => {
     const tree = renderer.create(<App />).toJSON();
     expect(tree).toMatchSnapshot();
   });
+
+  it('should initiate GET request using axios() to servlet', async () => {
+    const mockMatchStatus = {matchStatus: 'true'};
+    const mockData = {data: mockMatchStatus};
+    axios.get.mockResolvedValue(mockData);
+
+    const response = await fetchMatch();
+    return expect(response).toEqual(mockMatchStatus);
```

Comment:
Does the test need to return the result of the assertion? The pattern I've seen with tests that assert on promises is to return the promise at the end of the test. Like

```
 it('should initiate GET request using axios() to servlet', () => {
    const mockMatchStatus = {matchStatus: 'true'};
    const mockData = {data: mockMatchStatus};
    axios.get.mockResolvedValue(mockData);

    return fetchMatch().then((response) => {
        expect(response).toEqual(mockMatchStatus);
    })
});
```

would that work here or does that not work with this framework?

Reply:
Yes, both of these would work with the framework since both await and then resolve the promise.